packages/react-server/core/FramebackController.js

```diff
@@ -289,12 +289,15 @@ class FramebackController extends EventEmitter {
 		// If the frame has a client controller we'll listen to when
 		// _it_ tells us it's done loading.
 		//
+		if (clientController && !clientController.__parentIsListening) {
+			clientController.__parentIsListening = true;
+			clientController.context.onLoadComplete(this._handleFrameLoad.bind(this, frame));
+		}
+
 		// It may have finished _before_ we get here, so we'll also check
 		// whether it has already set its `_previouslyRendered` flag.
 		//
-		if (clientController && !clientController.__parentIsListening && !clientController._previouslyRendered) {
-			clientController.__parentIsListening = true;
-			clientController.context.onLoadComplete(this._handleFrameLoad.bind(this, frame));
+		if (clientController && !clientController._previouslyRendered) {
```

Comment:
> Can this be an else if block? or even an else block?

It can't, actually.  You'll notice that _previously_ the bodies of the two cases were _together_, and the conditionals were _anded_.  In the case where the frame's load event fires before the client controller finishes rendering we still need both to happen.